src/store/notifications/saga.test.ts

```diff
@@ -119,4 +129,134 @@ describe('notifications list saga', () => {
 
     expect(notificationsListResult).toEqual({ value: [] });
   });
+
+  it('watchForChannelEvent', () => {
+    const userId = 'user-id';
+
+    const notification = { id: 'notification-id', notificationType: sample(relevantNotificationTypes) };
+
+    const notificationChannel = {
+      take: jest.fn(),
+      close: jest.fn(),
+    };
+
+    testSaga(watchForChannelEvent, userId)
+      .next()
+      .call(createEventChannel, userId)
+      .next(notificationChannel)
+
+      .next({ abort: undefined, notification })
+      .call(addNotification, notification)
+      .next()
+      .inspect((raceValue) => {
+        expect(raceValue).toStrictEqual(
+          race({
+            abort: take(SagaActionTypes.CancelEventWatch),
+            notification: take(notificationChannel),
+          })
+        );
+      })
+
+      .next({ abort: true, notification: undefined })
+      .inspect((returnValue) => {
+        expect(notificationChannel.close).toBeCalled();
+        expect(returnValue).toBeFalse();
+      })
+
+      .next()
+      .isDone();
+  });
+
+  it('addNotification', async () => {
+    const notification = { id: 'id-added', name: 'this one should be added to the top of the list' };
+
+    const notificationsList = { value: ['id-old'] };
+    const notifications = { 'id-old': { id: 'id-old', name: 'this one already existed' } };
+
+    const {
+      storeState: {
+        normalized: { notifications: storeNotifications },
+        notificationsList: storeNotificationsList,
+      },
+    } = await expectSaga(addNotification, notification)
+      .withReducer(rootReducer)
+      .withState({
+        notificationsList,
+        normalized: { notifications },
+      })
+      .run();
+
+    expect(storeNotificationsList.value).toEqual([
+      notification.id,
+      ...notificationsList.value,
+    ]);
+    expect(storeNotifications).toStrictEqual({ ...{ [notification.id]: notification }, ...notifications });
+  });
+
+  it('authWatcher', async () => {
+    const channel = multicastChannel();
+    const userId = 'user-id';
+
+    testSaga(authWatcher)
+      .next()
+      .call(authChannel)
+
+      .next(channel)
+      .inspect((action) => {
+        const { payload } = action as any;
+
+        expect(payload.channel).toEqual(channel);
+        expect(payload.pattern).toEqual('*');
+      })
+
+      .next({ userId })
+      .spawn(watchForChannelEvent, userId)
+
+      .next(channel)
+      .inspect((action) => {
+        const { payload } = action as any;
+
+        expect(payload.channel).toEqual(channel);
+        expect(payload.pattern).toEqual('*');
+      })
+
+      .next({ userId: undefined })
+      .inspect((action) => {
+        expect(action).toEqual({ type: SagaActionTypes.CancelEventWatch });
+      })
+
+      .next();
+    // .isDone()
```

Comment:
What happens with this test if I omit `.isDone`? This test is a non breaking while loop.

Reply:
I've done it. I'm not entirely sure but even if the references to the unfinished generator hang around I can't imagine we'd over consume any memory or anything in our test suite. It'd take quite a bit, I think.